docs(1.9.0): important notes for v2 backing image incompatibility (#1102)
longhorn/website: content/docs/1.9.0/important-notes/_index.md

Deprecated APIs are no longer served and may therefore cause unexpected or unwanted behavior. Avoid using longhorn.io/v1beta1 in new code and, if possible, rewrite existing code to exclude this version.

## Breaking Change

### V2 Backing Image

> Due to naming conflicts in the extended attributes (xattrs) used by SPDK backing image logical volumes, the v1.9.0 V2 backing image is incompatible with previous versions.

Reviewer: ```suggestion
Due to naming conflicts in the extended attributes (xattrs) used by SPDK backing image logical volumes, the V2 backing image in version 1.9.0 and later is incompatible with earlier versions.
```

Deprecated APIs are no longer served and may therefore cause unexpected or unwanted behavior. Avoid using longhorn.io/v1beta1 in new code and, if possible, rewrite existing code to exclude this version.

## Breaking Change

### V2 Backing Image

> Due to naming conflicts in the extended attributes (xattrs) used by SPDK backing image logical volumes, the V2 backing image in version 1.9.0 and later is incompatible with earlier versions.

Reviewer: The relationship between the sentences and the workaround is not clear. Suggest
```suggestion
Due to naming conflicts in the extended attributes (xattrs) used by SPDK backing image logical volumes, the V2 backing image in version 1.9.0 and later is incompatible with earlier versions. To ......, you can apply the provided workaround.
```

> - **Before upgrading to v1.9.0**:

Reviewer: ```suggestion
- Before upgrading to v1.9.0:
```

> Due to naming conflicts in the extended attributes (xattrs) used by SPDK backing image logical volumes, the V2 backing image in version 1.9.0 and later is incompatible with earlier versions. The V2 backing images need to be deleted and recreated during the upgrade. Because backing image is undeletable before deleting the volumes on top of it, the volume backup, deletion, recreation, are also needed.

Reviewer 2: ```suggestion
Starting in Longhorn v1.9.0, V2 backing images are incompatible with earlier versions due to naming conflicts in the extended attributes (`xattrs`) used by SPDK backing image logical volumes. As a result, V2 backing images must be deleted and recreated during the upgrade process.

Since backing images cannot be deleted while volumes using them still exist, you must first back up, delete, and later restore those volumes.
```

>   - Ensure the backup targets are working.
>   - Create full backup for all volumes created using a V2 backing image.
>   - Detach and delete these volumes after backup.
>   - In backing image page, save all the spec of V2 backing images, including the name and all parameters.
>   - Delete all V2 backing image.

Reviewer: ```suggestion
  - Create backups for volumes using V2 backing images.
  - Detach and delete these volumes after backup.
  - Delete V2 backing images.
```

Author: Maybe we should emphasize that the backing image are going to be recreated using the same parameters here?

> - **After upgrading**:

Reviewer: ```suggestion
- After upgrading:
```

  - Recreate the necessary V2 backing images with the same name from the saved parameters.

>   - Wait for V2 backing image recreation. To each backing image, the number of ready copies should meet the minimum copies it requirs.

Reviewer: ```suggestion
  - Wait for V2 backing image recreation. To each backing image, the number of ready copies should meet the minimum copies it requires.
```

>   - Recreate the necessary V2 backing images with the same name from the saved parameters.

Reviewer: ```suggestion
  - Recreate the V2 backing images with the same names.
```

>   - Wait for V2 backing image recreation. To each backing image, the number of ready copies should meet the minimum copies it requires.

Reviewer: This might e redundant.
```suggestion
  - Wait for V2 backing image recreation. To each backing image, the number of ready copies should meet the minimum copies it requires.
```

>   - Restore and rebuild volumes from your backups.

Reviewer: ```suggestion
  - Restore volumes using the recreated backing images from the backups.
```

> - **Before upgrading to v1.9.0**:
>   - Ensure the backup targets are working.
>   - Create full backup for all volumes created using a V2 backing image.
>   - Detach and delete these volumes after backup.
>   - In backing image page, save all the spec of V2 backing images, including the name and all parameters.
>   - Delete all V2 backing image.
> - **After upgrading**:
>   - Recreate the necessary V2 backing images with the same name from the saved parameters.
>   - Wait for V2 backing image recreation. To each backing image, the number of ready copies should meet the minimum copies it requires.
>   - Restore and rebuild volumes from your backups.
> 
> For more information, see [#6534](https://github.com/longhorn/longhorn/issues/10805)

Reviewer 2: It should be Issue Number 10805 right?
```suggestion
#### Upgrade Instructions
**Before upgrading to v1.9.0**:
- Verify that backup targets are functioning properly.
- Create full backups of all volumes that use a V2 backing image.
- Detach and delete these volumes after the backups complete.
- In the **Backing Image** page, save the specifications of all V2 backing images (name and all parameters).
- Delete all V2 backing images.

**After upgrading**:
- Recreate the necessary V2 backing images using the saved specifications.
- Wait for each V2 backing image to be fully recreated. Ensure the number of ready replicas meets the required minimum.
- Restore the volumes from your backups and allow them to rebuild.

For more details, see [Issue #10805](https://github.com/longhorn/longhorn/issues/10805).
```

Author: Good catch, thanks!

Deprecated APIs are no longer served and may therefore cause unexpected or unwanted behavior. Avoid using longhorn.io/v1beta1 in new code and, if possible, rewrite existing code to exclude this version.

## Breaking Change

### V2 Backing Image

Starting with Longhorn v1.9.0, V2 backing images are incompatible with earlier versions due to naming conflicts in the extended attributes (`xattrs`) used by SPDK backing image logical volumes. As a result, V2 backing images must be deleted and recreated during the upgrade process.

Since backing images cannot be deleted while volumes using them still exist, you must first back up, delete, and later restore those volumes.

#### Upgrade Instructions

- Before upgrading to v1.9.0:
  - Verify that backup targets are functioning properly.
  - Create full backups of all volumes that use a V2 backing image.
  - Detach and delete these volumes after the backups complete.
  - In the **Backing Image** page, save the specifications of all V2 backing images, including the name and the image source.
  - Delete all V2 backing images.
- After upgrading:

>   - Recreate the V2 backing images using the saved name and image source.

Reviewer: ```suggestion
  - Recreate the V2 backing images using the same names and image sources.
```

>   - Restore the volumes from your backups and allow them to rebuild.

Reviewer: ```suggestion
  - Restore the volumes from your backups and corresponding backing images.
```

Author: It does not ask for a backing image while restoring a volume from UI. The backing image is required by the backup rebuilding internally.

Reviewer: If it is internal rebuilding, we don't need to mention `allow them to rebuild`.

Deprecated APIs are no longer served and may therefore cause unexpected or unwanted behavior. Avoid using longhorn.io/v1beta1 in new code and, if possible, rewrite existing code to exclude this version.

## Breaking Change

### V2 Backing Image

The v1.9.0 V2 backing image is incompatible with previous versions.

> - Before upgrade to v1.9.0, backup all the volumes that are built from a V2 backing image, then remove these volumes.
> - After upgrade to v1.9.0, recreate all V2 backing images, and rebuild the volumes from the backup.

Reviewer 2: ```suggestion
- **Before upgrading to v1.9.0**:
  - Backup all volumes created using a V2 backing image.
  - Delete these volumes after backup.
- **After upgrading**:
  - Recreate the necessary V2 backing images.
  - Restore and rebuild volumes from your backups.
```

Deprecated APIs are no longer served and may therefore cause unexpected or unwanted behavior. Avoid using longhorn.io/v1beta1 in new code and, if possible, rewrite existing code to exclude this version.

## Breaking Change

### V2 Backing Image

> Due to SPDK logical volume xattr schema change, the v1.9.0 V2 backing image is incompatible with previous versions.

Reviewer: ```suggestion
Due to naming conflicts in the extended attributes (xattrs) used by SPDK backing image logical volumes, the v1.9.0 V2 backing image is incompatible with previous versions.
```

- **Before upgrading to v1.9.0**:
  - Backup all volumes created using a V2 backing image.
  - Delete these volumes after backup.
- **After upgrading**:
  - Recreate the necessary V2 backing images.
  - Restore and rebuild volumes from your backups.

> For more information, see [#6534](https://github.com/longhorn/longhorn/issues/10805)

Reviewer: Can you mention that the xattr naming conflict causes the incompatibility?

Author: Added into the line. Please help review again, thanks.

>   - Restore and rebuild volumes from your backups.

Reviewer: @COLDTURNIP Please check if the workaround is valid before merging the PR. Thanks.

### Longhorn System Upgrade

> Longhorn currently does not support live upgrading of V2 volumes and backing images. Ensure that all V2 volumes are detached before initiating the upgrade process. To volumes created from a V2 backing image, backup the volumes before upgrading Longhorn.

Reviewer 2: ```suggestion
Longhorn currently does not support live upgrading of V2 volumes or V2 backing images.

Before starting the upgrade process:
- Detach all V2 volumes from their workloads.
- Backup any volumes created from V2 backing images to avoid data loss.
```

### Longhorn System Upgrade

Longhorn currently does not support live upgrading of V2 volumes or V2 backing images. Before starting the upgrade process:

- Detach all V2 volumes from their workloads.

> - Backup any volumes created from V2 backing images to avoid data loss.

Reviewer: The change is redundant because you have mentioned the backing image issue before.
